src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -647,42 +648,182 @@ public void Fault(RequestChannel channel)
                             throw;
                         }
                     }
+                }
 
-                    try
+                private bool PrepareMessageHeaders(Message message)
```

Comment:
This code came from various methods in HttpInput placed into the correct order here.